thrust/thrust/zip_function.h

```diff
@@ -154,6 +140,9 @@ template <typename Function>
 class zip_function
 {
 public:
+  //! Default constructs Function.
```

Comment:
I hate that too, so what do you prefer:
* Drop the comment
* Reword to "Default constructs the contained function object"

I thought I would make it clear that there is no partially formed state, and you can call a default constructed `zip_function`.

Also, I spelled it `Function` on purpose, because it is the name of the class template parameter. `function` does not refer to any entity within this scope.

Reply:
"Default constructs the contained function object." it is.